gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource.traversal;
+import static org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__.hasLabel;
+import static org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__.inE;
+import static org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__.outE;
+import static org.junit.Assert.assertEquals;
+
+public class GolangTranslatorTest {
+    private static final GraphTraversalSource g = traversal().withEmbedded(EmptyGraph.instance());
+    private static final Translator.ScriptTranslator translator = GolangTranslator.of("g");
+
+    @Test
+    public void shouldHaveValidToString() {
+        assertEquals("translator[h:gremlin-go]", GolangTranslator.of("h").toString());
+    }
+
+    @Test
+    public void shouldTranslateOption() {
+        final String gremlinAsGo = translator.translate(
+                g.V().has("person", "name", "marko").asAdmin().getBytecode()).getScript();
+        assertEquals("g.V().Has(\"person\", \"name\", \"marko\")", gremlinAsGo);
+    }
+
+    @Test
+    public void shouldTranslateCardinality() {
+        final String gremlinAsGo = translator.translate(
+                g.addV("person").property(VertexProperty.Cardinality.list, "name", "marko").asAdmin().getBytecode()).getScript();
+        assertEquals("g.AddV(\"person\").Property(Cardinality.List, \"name\", \"marko\")", gremlinAsGo);
+    }
+
+    @Test
+    public void shouldTranslateMultilineStrings() {
+        final String gremlinAsGo = translator.translate(
+                g.addV().property("text", "a" + System.lineSeparator() + "\"and\"" + System.lineSeparator() + "b").asAdmin().getBytecode()).getScript();
+        final String escapedSeparator = StringEscapeUtils.escapeJava(System.lineSeparator());
+        final String expected = "g.AddV().Property(\"text\", \"a" + escapedSeparator + StringEscapeUtils.escapeJava("\"and\"") + escapedSeparator + "b\")";
+        assertEquals(expected, gremlinAsGo);
+    }
+
+    @Test
+    public void shouldTranslateChildTraversals() {
+        final String gremlinAsGo = translator.translate(
+                g.V().has("person", "name", "marko").
+                        where(outE()).asAdmin().getBytecode()).getScript();
+        assertEquals("g.V().Has(\"person\", \"name\", \"marko\").Where(gremlingo.AnonTrav__.OutE())", gremlinAsGo);
+    }
+
+    @Test
+    public void shouldTranslatePythonNamedSteps() {
+        final String gremlinAsGo = translator.translate(
+                g.V().has("person", "name", "marko").
+                        where(outE().count().is(2).and(__.not(inE().count().is(3)))).asAdmin().getBytecode()).getScript();
+        assertEquals("g.V().Has(\"person\", \"name\", \"marko\").Where(gremlingo.AnonTrav__.OutE().Count().Is(2).And(gremlingo.AnonTrav__.Not(gremlingo.AnonTrav__.InE().Count().Is(3))))", gremlinAsGo);
+    }
```

Comment:
Have you run these / do they pass?

Reply:
Yes, the tests pass except for the disabled ones.